tests/assessment/test_validity_checks.py

```diff
@@ -0,0 +1,230 @@
+from gnomad.assessment.validity_checks import (
+    check_array_struct_missingness,
+    check_missingness_of_struct,
+    flatten_missingness_struct,
+    unfurl_array_annotations,
+)
+
+
+@pytest.fixture
+def ht_for_check_missingness_of_struct():
+    """Fixture to set up a Hail Table with the desired nested structure and data."""
+    # Create test data with nested structs.
+    data = [
+        hl.struct(
+            idx=0,
+            s=hl.struct(
+                a=1,
+                b="value1",
+                c=hl.struct(d=[hl.missing(hl.tstr), hl.missing(hl.tstr)], e="test1"),
+            ),
+        ),
+        hl.struct(
+            idx=0,
+            s=hl.struct(
+                a=2,
+                b="value2",
+                c=hl.struct(
+                    d=["not missing", hl.missing(hl.tstr)], e=hl.missing(hl.tstr)
+                ),
+            ),
+        ),
+        hl.struct(
+            idx=0,
+            s=hl.struct(
+                a=3,
+                b=hl.missing(hl.tstr),
+                c=hl.struct(d=hl.missing(hl.tarray(hl.tstr)), e=hl.missing(hl.tstr)),
+            ),
+        ),
+        hl.struct(
+            idx=0,
+            s=hl.struct(a=4, b="value3", c=hl.struct(d=["foo", "bar"], e="test2")),
+        ),
+        hl.struct(
+            idx=0,
+            s=hl.struct(
+                a=5, b="value4", c=hl.struct(d=hl.empty_array(hl.tstr), e="test3")
+            ),
+        ),
+    ]
+
+    # Convert data into a Hail table.
+    ht = hl.Table.parallelize(
+        data,
+        hl.tstruct(
+            idx=hl.tint32,
+            s=hl.tstruct(
+                a=hl.tint32, b=hl.tstr, c=hl.tstruct(d=hl.tarray(hl.tstr), e=hl.tstr)
+            ),
+        ),
+    )
+
+    return ht
+
+
+def test_check_missingness_of_struct(ht_for_check_missingness_of_struct):
+    """Test check_missingness_of_struct and flatten results."""
+    ht = ht_for_check_missingness_of_struct
+
+    # Apply check_missingness_of_struct aggregate results.
+    metric_missingness = check_missingness_of_struct(ht.s, "s")
+    missingness_struct = ht.aggregate(hl.struct(**metric_missingness))
+
+    # Flatten the result struct into a dictionary.
+    missingness_dict = flatten_missingness_struct(missingness_struct)
+
+    # Define expected missingness results.
+    expected_result = {
+        "s.a": 0.0,
+        "s.b": 0.2,
+        "s.c.d": 0.6,
+        "s.c.e": 0.4,
+    }
+
+    # Compare the results with the expected values.
+    for key, expected_value in expected_result.items():
+        assert missingness_dict[key] == expected_value
+
+
+@pytest.fixture()
+def ht_for_check_array_struct_missingness():
+    """Fixture to set up a sample Hail Table with array<struct> fields and global index dictionary ."""
+    # Define rows of the Table.
+    data = [
+        {
+            "idx": 1,
+            "freq": [
+                {"AC": 10, "AF": 0.1, "AN": 20},
+                {"AC": 5, "AF": 0.05, "AN": None},
```

Comment:
I know it's just for testing, but it's funny to see defined ACs/AFs without ANs 😆